warehouse/accounts/services.py

```diff
@@ -586,6 +587,7 @@ def database_login_factory(context, request):
     return DatabaseUserService(
         request.db,
         metrics=request.find_service(IMetricsService, context=None),
+        remote_addr=request.remote_addr,
```

Comment:
Here's the key change in this refactor: the `remote_addr` is passed in when creating the `DatabaseUserService`, instead of being passed to every function on the `DatabaseUserService`.